torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -300,14 +340,56 @@ void ProcessGroupNCCL::ncclCommWatchdog() {
             // waiting for the operation to complete.
             for (const auto& ncclComm : ncclComms) {
               ncclComm->ncclCommAbort();
+              abortedCommIds.emplace(
+                  buildNcclUniqueIdStr(ncclComm->getNcclId()));
             }
           }
+        }
+      }
+    }
 
-          // Remove communicators from the cache.
-          it = devNCCLCommMap_.erase(it);
```

Comment:
would you please also explain why we do not need to erase the cache? IIUC, since all processes will abort ncclComm, application codes should be able to catch exception and destroy process group?